sdk/typescript/src/multisig/publickey.ts

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { fromB64, toB64 } from '@mysten/bcs';
+import { blake2b } from '@noble/hashes/blake2b';
+import { bytesToHex } from '@noble/hashes/utils';
+import { PublicKey, bytesEqual } from '../cryptography/publickey.js';
+import type {
+	SerializedSignature,
+	SignatureFlag,
+	SignatureScheme,
+} from '../cryptography/signature.js';
+import {
+	SIGNATURE_FLAG_TO_SCHEME,
+	SIGNATURE_SCHEME_TO_FLAG,
+	parseSerializedSignature,
+} from '../cryptography/signature.js';
+import { normalizeSuiAddress } from '../utils/sui-types.js';
+import { builder } from '../builder/bcs.js';
+// eslint-disable-next-line import/no-cycle
+import { publicKeyFromBytes } from '../verify/index.js';
+import { bcs } from '../types/sui-bcs.js';
+
+type CompressedSignature =
+	| { ED25519: number[] }
+	| { Secp256k1: number[] }
+	| { Secp256r1: number[] };
+
+type PublicKeyEnum = { ED25519: number[] } | { Secp256k1: number[] } | { Secp256r1: number[] };
+
+type PubkeyEnumWeightPair = {
+	pubKey: PublicKeyEnum;
+	weight: number;
+};
+
+type MultiSigPublicKeyStruct = {
+	pk_map: PubkeyEnumWeightPair[];
+	threshold: number;
+};
+
+export type MultiSigStruct = {
+	sigs: CompressedSignature[];
+	bitmap: number;
+	multisig_pk: MultiSigPublicKeyStruct;
+};
+
+type ParsedPartialMultiSigSignature = {
+	signatureScheme: SignatureScheme;
+	signature: Uint8Array;
+	publicKey: PublicKey;
+	weight: number;
+};
+
+export const MAX_SIGNER_IN_MULTISIG = 10;
+
+/**
+ * A MultiSig public key
+ */
+export class MultiSigPublicKey extends PublicKey {
+	private rawBytes: Uint8Array;
+	private multisigPublicKey: MultiSigPublicKeyStruct;
+	private publicKeys: {
+		weight: number;
+		publicKey: PublicKey;
+	}[];
+	/**
+	 * Create a new MultiSigPublicKey object
+	 * @param value MultiSig public key as buffer or base-64 encoded string
```

Comment:
I prefer defining JSDocs inline on the param directly so that it's less likely to get out-of-sync